### PR TITLE
Fix transparent background in some screen scaling modes

### LINKED ISSUE #4123: With Screen Scaling: Excluding Overlays, Everything: Background dimming makes background transparent instead
**Describe your problem:** When using screen scaling within the modes "excluding overlays" and "everything" the background dimming option does not dimm the background but makes it transparent instead

**Screenshots or videos showing encountered issue:** [screen](https://puu.sh/CAKnO/fd28b97148.png)

**osu!lazer version:** 2019.122.1

### PR DESCRIPTION
**Pull request by chrisny286**
by replacing `.FadeTo` at the necessary places with `.FadeColour`.
restoring the `.FadeColour` in `fadeOut` of the Player Screen which is ran in every exit scenario.
- Closes #4123

---

Fix transparent backgrounds in some screen scaling modes

### CODE REVIEW

File: osu.Game/Screens/Play/Player.cs [hunk 1]
```diff
@@ -438,7 +439,7 @@ protected override void UpdateBackgroundElements()
                 .FadeTo(storyboardVisible && BackgroundOpacity > 0 ? 1 : 0, BACKGROUND_FADE_DURATION, Easing.OutQuint);
 
             if (storyboardVisible && beatmap.Storyboard.ReplacesBackground)
-                Background?.FadeTo(0, BACKGROUND_FADE_DURATION, Easing.OutQuint);
+                Background?.FadeColour(OsuColour.Gray(0f), BACKGROUND_FADE_DURATION, Easing.OutQuint);
```
Comment (peppy) [inline osu.Game/Screens/Play/Player.cs]: Color4.Black

File: osu.Game/Screens/Play/Player.cs [hunk 2]
```diff
@@ -401,6 +401,7 @@ private void fadeOut(bool instant = false)
         {
             float fadeOutDuration = instant ? 0 : 250;
             Content.FadeOut(fadeOutDuration);
+            Background?.FadeColour(OsuColour.Gray(1f), fadeOutDuration, Easing.OutQuint);
```
Comment (peppy) [inline osu.Game/Screens/Play/Player.cs]: Color4.White

**Review by peppy — CHANGES_REQUESTED** [review]

File: osu.Game/Screens/Play/Player.cs (hunk 1 above)
Comment (chrisny286) [reply osu.Game/Screens/Play/Player.cs]: would need to add another using osuTK.Graphics for this too since its not needed in this file yet

Comment (peppy) [reply osu.Game/Screens/Play/Player.cs]: correct

**Comment by peppy** [comment]
You missed a usage on line 167 of `PlayerLoader`.

**Comment by chrisny286** [comment]
in Player Loader it does not matter because the value which is faded to is 1 > solid background

**Comment by peppy** [comment]
It is required. `PlayerLoader` changes to and from the user dim on hover/hoverlost. Check again.

**Review by peppy — APPROVED** [review]

**Comment by peppy** [comment]
This looks fine now.

As a note, there's going to come a point we probably want to fix this properly, by moving the use opacity logic *inside* the `BackgroundScreenBeatmap` class, so we don't have fighting for colour changes at when it's modified externally.

**Comment by chrisny286** [comment]
so that the `Player` class only needs to call something like  "toggleDim()" method ?

**Comment by peppy** [comment]
i'd make it explicitly `EnableUserDim` or something, yeah.

**Pull request merged**